tiledb/api/src/array/enumeration/strategy.rs

```diff
@@ -0,0 +1,161 @@
+use std::cmp::Ordering;
+
+use proptest::collection::vec;
+use proptest::prelude::*;
+
+use crate::array::EnumerationData;
+use crate::datatype::strategy::*;
+use crate::{fn_typed, Datatype};
+
+pub fn prop_enumeration_name() -> impl Strategy<Value = String> {
+    proptest::string::string_regex("[a-zA-Z0-9_]+")
+        .expect("Error creating attribute name strategy")
+}
+
+fn prop_cell_val_num() -> impl Strategy<Value = Option<u32>> {
+    Just(None)
+}
+
+fn prop_ordered() -> impl Strategy<Value = bool> {
+    any::<bool>()
+}
+
+trait WtfNumCmp: std::marker::Sized {
```

Comment:
Is NaN allowed in enumerations? I was also curious why the NotNan trait you showed me was of no use here.

Reply:
NaN is passable yeah. Inside enumerations we don't do any interpretation of the bytes given by the user. So long as they are the correct number of bytes for each value we just store them and spit them back out. So theoretically, multiple NaN's are even possible. So long as the bit patterns are distinct is all we care about internally.

TotalOrder looks like part of making this solution simpler. I keep forgetting about blanket trait implementations so that should just turn this into a much simpler trait.

Reply:
Oh, I meant my first paragraph as a lamentation, I think we speciifcally cannot use it because we need it for something other than floats, and the crate didn't give us that support.  Unless you want to call the implementation inside of what you've done here.

Reply:
Fixed. I linked this earlier, but I figure I should paste a link to the final implementation for posterity.

https://github.com/TileDB-Inc/tiledb-rs/blob/2d5c2940affe22ee36175f424bb5cb5bdccac9ba/tiledb/api/src/util/strategy.rs